tools/clang/include/clang/SPIRV/SpirvType.h

```diff
@@ -240,17 +246,28 @@ class RuntimeArrayType : public SpirvType {
     return t->getKind() == TK_RuntimeArray;
   }
 
-  const SpirvType *getElementType() const { return elementType; }
+  const SpirvType *getElementType() const {
+    return llvm::cast<SpirvType>(elementType);
+  }
 
 private:
   const SpirvType *elementType;
 };
 
 class StructType : public SpirvType {
 public:
-  StructType(llvm::ArrayRef<const SpirvType *> memberTypes,
-             llvm::StringRef name, llvm::ArrayRef<llvm::StringRef> memberNames,
-             bool isReadOnly);
+  enum class InterfaceType : uint32_t {
+    NonInterface = 0,
+    Ssbo = 1,
+    NonSsbo = 2,
+  };
+
+  StructType(
+      llvm::ArrayRef<const SpirvType *> memberTypes, llvm::StringRef name,
```

Comment:
Hmm, with so many separate ArrayRefs, it becomes a little bit cumbersome to handle. What about creating a
```c++
struct FieldInfo {
  const SpirvTpye* type;
  std::string name;
  ...
};
```

Reply:
Done